### PR TITLE
Mine 5.1.0.2 schema plus massive reorg of file structure

### PR DESCRIPTION
**Pull request by sammyjava**
This is the Big PR, with the schema based on the LIS mine data model 5.1.0.2 and the massive reorganization of files and includes and all the magic that @alancleary implemented to reduce the overhead of adding new types, resolvers, API methods, etc.

The plan is to use this GraphQL server for our first production implementation of web-component queries on jekyll-legumeinfo (once LegumeMine finishes the current build so we've got a 5.1.0.2 LegumeMine in place).

### CODE REVIEW

**Comment by sammyjava** [comment]
Oh, another important addition is the MineWebProperties type which is populated from the mine web-properties REST call. This should be used to check the graphql-server mine version against that of the mine being queried.

File: src/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+├── intermine
+│   ├── expression-sample.js
+│   ├── expression-source.js
+│   ├── gene-family-assignment.js
+│   ├── gene-family.js
+│   ├── gene.js
+│   ├── ...
+```
+
+Right now, the scripts under `data-sources` are not yet broken apart, but I (Sam) plan to do so.
```
Comment (alancleary) [inline src/README.md:31]: It looks like the scripts in `data-sources` have been broken apart. If so, can you remove this line?

File: src/types/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    "citation": "<a href=\"http://www.ncbi.nlm.nih.gov/pubmed/23023984\" target=\"_blank\">Smith RN, et al. InterMine: a flexible data warehouse system for the integration and analysis of heterogeneous biological data. Bioinformatics. 2012 Dec 1;28(23):3163-5.</a>",
+    "releaseVersion": "5.1.0.2",
+    "standalone": "true",
+    "helpLocation": "https://mines.legumeinfo.org/minimine/help",
+    "title": "MiniMine",
+    "sitePrefix": "https://mines.legumeinfo.org/minimine"
+}
+```
+
+## classes populated in LIS InterMine 5.1.0.2 (excluding simple objects)
```
Comment (alancleary) [inline src/types/README.md:34]: Thanks for doing this. It's very helpful!

**Review by alancleary — APPROVED** [review]
Looks good to me. We can implement the Intermine version check outside of this PR.

**Pull request merged**